debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsStreamProducer.java

```diff
@@ -389,7 +387,8 @@ protected void generateDeleteRecord(TableId tableId, Object[] oldRowData, Blocki
         assert tableSchema != null;
         Object key = tableSchema.keyFromColumnData(oldRowData);
         Struct value = tableSchema.valueFromColumnData(oldRowData);
-        if (key == null || value == null) {
+        if (value == null) {
+            logger.warn("ignoring message from delete transaction for table '{}' because it does not have a primary key defined and replica identity for the table is not FULL", tableId);
```

Comment:
Again, delete from any table without PK but with replica identity FULL will produce message without key. But since message contains all required data (in 'before' part) it can be processed even without PK.

Also I agree with "transaction", I should probably change it to the "message" or "event".